mdal/frmts/mdal_xmdf.hpp

```diff
@@ -20,13 +20,56 @@
 namespace MDAL
 {
 
+  /**
+   * The XmdfDataset reads the data directly from HDF5 file
+   * by usage of hyperslabs retrieval
+   *
+   * basically all (timesteps) data for one particular dataset groups
+   * are stored in single 2D array with structure
+   *
+   * timestep1 x1 y1 x2 y2 ...... xN yN
+   * timestep2 x1 y1 x2 y2 ...... xN yN
+   * ...
+   * timestepN x1 y1 x2 y2 ...... xN yN
+   *
+   * or for scalar datasets
+   *
+   * timestep1 x1 x2 ...... xN
+   * timestep2 x1 x2 ...... xN
+   * ...
+   * timestepN x1 x2 ...... xN
+   */
+  class XmdfDataset: public Dataset
+  {
+    public:
+      XmdfDataset( const HdfDataset &valuesDs,
+                   const HdfDataset &activeDs,
+                   hsize_t timeIndex );
+      ~XmdfDataset() override;
+
+      size_t scalarData( size_t indexStart, size_t count, double *buffer ) override;
+      size_t vectorData( size_t indexStart, size_t count, double *buffer ) override;
+      size_t activeData( size_t indexStart, size_t count, char *buffer ) override;
+
+      const HdfDataset &dsValues() const;
+      const HdfDataset &dsActive() const;
+      hsize_t timeIndex() const;
+
+    private:
+      HdfDataset mHdf5DatasetValues;
+      HdfDataset mHdf5DatasetActive;
```

Comment:
are these some simple pointers? I'm wondering if keeping them open for each dataset is a good practice, i.e. if they do not force hdf library to consume some extra resources (even though we may not need to used those datasets)

Reply:
they are shared pointers, and there is one pointer for ALL timesteps in one group, so basically we should not run to exotic numbers here